jmbitcoin/jmbitcoin/secp256k1_transaction.py

```diff
@@ -388,57 +415,141 @@ def address_to_script(addr):
 # Output script to address representation
 
 def is_p2pkh_script(script):
-    if script[:3] == b'\x76\xa9\x14' and script[-2:] == b'\x88\xac' and len(
+    """ Given a script as bytes, returns True if the script
+    matches the pay-to-pubkey-hash pattern (using opcodes),
+    otherwise returns False.
+    """
+    if script[:3] == P2PKH_PRE and script[-2:] == P2PKH_POST and len(
             script) == 25:
         return True
     return False
 
 def is_segwit_native_script(script):
-    """Is scriptPubkey of form P2WPKH or P2WSH"""
-    if script[:2] in [b'\x00\x14', b'\x00\x20']:
+    """Is script, as bytes, of form P2WPKH or P2WSH;
+    see BIP141 for definitions of 2 current valid scripts.
+    """
+    if not isinstance(script, bytes):
+        script = binascii.unhexlify(script)
```

Comment:
Why do some is_*_script functions have a hex check and others don't? Ideally we would get rid of those checks and only handle non-hex data.

Reply:
Good point. Occasionally there are calls directly from non-wallet code e.g. `is_native_segwit_script` is called by taker in a hex context. Most other calls from outside jmbitcoin are from the wallet code, which is all binary. I've added the above check to the other functions that can be called from outside the package, in